pipestat/pipestat.py

```diff
@@ -460,6 +462,144 @@ def summarize(
         report_path = html_report_builder(pipeline_name=self.pipeline_name, amendment=amendment)
         return report_path
 
+    @require_backend
+    def table(
+        self,
+        pipeline_type: Optional[str] = None,
+    ) -> List[str]:
+        """
+        Generates stats and object tables as .tsv file and .yaml files respectively.
+        :param Optional[str] pipeline_type: sample or project pipeline
+        :return list[str] table_path_list: list containing output file paths of stats and objects
+
+        """
+
+        pipeline_name = self.pipeline_name
+        pipeline_type = pipeline_type or self[PIPELINE_TYPE]
+        stats_path = self._create_stats_summary(pipeline_name, pipeline_type)
+        objs_path = self._create_obj_summary(pipeline_name, pipeline_type)
+
+        table_path_list = [stats_path, objs_path]
+        return table_path_list
+
+    def _create_stats_summary(self, pipeline_name, pipeline_type) -> str:
+        """
+        Create stats spreadsheet and columns to be considered in the report, save
+        the spreadsheet to file
+
+        :param pipestat manager object: the project to be summarized
+        :param str pipeline_name: name of the pipeline to tabulate results for
+        :param str pipeline_type: whether the project-level pipeline results
+            should be tabulated
+        :return str: tsv_outfile_path
+        """
+        # Create stats_summary file
+        columns = set()
+        stats = []
+        _LOGGER.info("Creating stats summary")
+        if pipeline_type == "project":
+            # TODO for project level should we actually be calling project.backend.get_samples(pipeline_type='project')?
+            reported_stats = ["project_name", self.project_name or "No Project Name Supplied"]
+            results = fetch_pipeline_results(
+                project=self,
+                pipeline_name=pipeline_name,
+                sample_name=None,
+                inclusion_fun=lambda x: x not in OBJECT_TYPES,
+                pipeline_type=pipeline_type,
+            )
+            for k, v in results.items():
+                reported_stats.append(k)
+                reported_stats.append(v)
+            stats.append(reported_stats)
+            columns = ["Project", "Project Name", "Results"]
+
+        else:
+            sample_index = 0
+            for sample in self.backend.get_samples(pipeline_type="sample"):
+                sample_index += 1
+                sample_name = sample[0]
+                pipeline_type = sample[1]
+                reported_stats = [sample_index, sample_name]
+                results = fetch_pipeline_results(
```

Comment:
I see both of these `_fetch` functions call this `fetch_pipeline_results` function. Also, the two functions are called right after one another.

Might it be more efficient to call this once, outside these calls, and pass the result to each of these sub-functions?